include/oneapi/dpl/pstl/execution_impl.h

```diff
@@ -29,100 +30,134 @@ namespace dpl
 namespace __internal
 {
 
-/* predicate */
+//------------------------------------------------------------------------
+// backend selector with tags
+//------------------------------------------------------------------------
+
+#if _ONEDPL_PAR_BACKEND_TBB
+using __par_backend_tag = __tbb_backend_tag;
+#elif _ONEDPL_PAR_BACKEND_OPENMP
+using __par_backend_tag = __omp_backend_tag;
+#elif _ONEDPL_PAR_BACKEND_SERIAL
+using __par_backend_tag = __serial_backend_tag;
+#else
+#    error "Parallel backend was not specified"
+#endif
+
+template <class _IsVector>
+struct __serial_tag
+{
+    using __is_vector = _IsVector;
+};
 
-template <typename _Tp>
-::std::false_type __lazy_and(_Tp, ::std::false_type)
+template <class _IsVector>
+struct __parallel_tag
```

Comment:
I prefer `__parallel_random_tag` as this is more clear about what tag represents in contrast to `__parallel_forward_tag`.  Specifically I think understanding the difference between `__parallel_forward_tag` and `__parallel_[random_]tag<std::false_type>` is lacking without the "random" in the name.  

I know this may cause some issues with upstream...

Reply:
My proposal - to save this name as is.